lib/src/apigateway/openapi-gateway-lambda.ts

```diff
@@ -219,17 +239,15 @@ export const generateOpenapiDocWithExtensions = (
   for (let i = 0; i < props.openapiOperations.length; i += 1) {
     const lambdaOperation = props.openapiOperations[i];
 
-    registry.registerPath(lambdaOperation.routeConfig);
-    let { operationId } = lambdaOperation.routeConfig;
-    if (!operationId) {
-      operationId = generateOperationId(lambdaOperation.routeConfig);
-      lambdaOperation.routeConfig.operationId = operationId;
-    }
-    if (operationId) {
-      lambdaOperationsMap[operationId] = lambdaOperation;
-    } else {
-      throw new Error("'operationId' should be defined");
-    }
+    const operationId =
+      lambdaOperation.routeConfig.operationId ?? generateOperationId(lambdaOperation.routeConfig);
+
+    registry.registerPath({
+      ...lambdaOperation.routeConfig,
+      operationId,
+    });
+
+    lambdaOperationsMap[operationId] = lambdaOperation;
```

Comment:
I changed this code block because as far as I understand we don't need to mutate the `lambdaOperation.routeConfig.operationId` property, it doesn't seems to be used anywhere else.

Previously it was mutating it, and as objects are passed as references it was impacting the caller of our construct.

```ts
declare const lambdaAlias: IAlias;

const openapiOperations = [
  {
    lambdaAlias,
    routeConfig: {
      path: '/user',
      method: 'post',
    },
  },
];

new OpenApiGatewayLambda(stack, 'myapi', {
  stage: 'tst',
  openapiBasic: {
  openapi: '3.0.3',
    info: {
      title: 'test api',
      version: 'v1',
    },
  },
  openapiOperations,
});

// Now the `openapiOperations` arrays will have the `operationId` for each operation
/**
[
  {
    lambdaAlias,
    routeConfig: {
      path: '/user',
      method: 'post',
      operationId: 'post-user-random5',
    },
  },
]
*/
```

This was also making the unit tests changes the behaviour depending of the order it was ran.
As we are reusing this routConfig object through the test cases, the first one to run would modify the object and the others would have the `lambdaOperation.routeConfig.operationId` unexpectedly filled in.

Reply:
Ok, if the ZodOpenAPI doesn't need it set this way (maybe when we register in the registry using the operationId it already does the trick), I think there is no problem and it's a good improvement to the code itself :)

Reply:
Yeah, if you look [this line](https://github.com/flaviostutz/cdk-practical-constructs/blob/22fd916df35eed3d9c3e2c0d2e8161cff12714ac/lib/src/apigateway/openapi-gateway-lambda.ts#L235-L236) it only uses the registry to generate the open api document.